src/video_core/shader/shader_ir.cpp

```diff
@@ -387,9 +387,49 @@ void ShaderIR::SetInternalFlagsFromInteger(NodeBlock& bb, Node value, bool sets_
     if (!sets_cc) {
         return;
     }
-    Node zerop = Operation(OperationCode::LogicalIEqual, std::move(value), Immediate(0));
-    SetInternalFlag(bb, InternalFlag::Zero, std::move(zerop));
-    LOG_WARNING(HW_GPU, "Condition codes implementation is incomplete");
+    switch (value->index()) {
+    case 0:
+        Iterop(bb, value);
+        break;
+    case 2:
+        if (const auto gpr = std::get_if<GprNode>(value.get())) {
+            LOG_WARNING(HW_GPU, "GprNode: index={}", gpr->GetIndex());
+            Node zerop = Operation(OperationCode::LogicalIEqual, std::move(value),
+                                   Immediate(gpr->GetIndex()));
+            SetInternalFlag(bb, InternalFlag::Zero, std::move(zerop));
+        }
+        break;
+
+    default:
+        Node zerop = Operation(OperationCode::LogicalIEqual, std::move(value), Immediate(0));
+        SetInternalFlag(bb, InternalFlag::Zero, std::move(zerop));
+        LOG_WARNING(HW_GPU, "Node Type: {}", value->index());
+        break;
+    }
+}
+
+void ShaderIR::Iterop(NodeBlock& nb, Node var) {
+    if (const auto op = std::get_if<OperationNode>(var.get())) {
```

Comment:
We generally use guard clauses to reduce nesting where possible:

```suggestion
    const auto* op = std::get_if<OperationNode>(var.get());
    if (op == nullptr) {
        return;
    }
    
    if (op->GetOperandsCount() == 0) {
        return;
    }
    
    // rest of the code here
```